File: server/dotnet/sdk-dotnet-example/Business/Domain/Payments/PaymentDetails/CardPaymentMethodSpecificOutput.cs
namespace Business.Domain.Payments.PaymentDetails;

public class CardPaymentMethodSpecificOutput
{
    public AcquirerInformation? AcquirerInformation { get; set; }

    public string? AuthorisationCode { get; set; }

    public CardEssentials? Card { get; set; }

    public CardFraudResults? FraudResults { get; set; }

    public string? PaymentAccountReference { get; set; }

    public int? PaymentProductId { get; set; }

    public ThreeDSecureResults? ThreeDSecureResults { get; set; }

    public string? InitialSchemeTransactionId { get; set; }
    
    public string? SchemeReferenceData { get; set; }
    
    public string? Token { get; set; }
    
    public string? PaymentOption { get; set; }
    
    public ExternalTokenLinked? ExternalTokenLinked { get; set; }
    
    public long? AuthenticatedAmount { get; set; }
    
    public CurrencyConversion? CurrencyConversion { get; set; }
    
    public PaymentProduct3208SpecificOutput? PaymentProduct3208SpecificOutput { get; set; }
    
    public PaymentProduct3209SpecificOutput? PaymentProduct3209SpecificOutput { get; set; }
    
    public ReattemptInstructions? ReattemptInstructions { get; set; }
}

File: server/python/app/application/domain/payments/payment_details/card_payment_method_specific_output.py
from dataclasses import dataclass
from typing import Optional, Any

from app.application.domain.payments.payment_details.acquirer_information import AcquirerInformation
from app.application.domain.payments.payment_details.card_essentials import CardEssentials
from app.application.domain.payments.payment_details.card_fraud_results import CardFraudResults
from app.application.domain.payments.payment_details.currency_conversion import CurrencyConversion
from app.application.domain.payments.payment_details.external_token_linked import ExternalTokenLinked
from app.application.domain.payments.payment_details.payment_product_3208_specific_output import PaymentProduct3208SpecificOutput
from app.application.domain.payments.payment_details.payment_product_3209_specific_output import PaymentProduct3209SpecificOutput
from app.application.domain.payments.payment_details.reattempt_instructions import ReattemptInstructions
from app.application.domain.payments.payment_details.three_d_secure_results import ThreeDSecureResults

@dataclass
class CardPaymentMethodSpecificOutput:
    acquirer_information: Optional[AcquirerInformation] = None
    authorisation_code: Optional[str] = None
    card: Optional[CardEssentials] = None
    fraud_results: Optional[CardFraudResults] = None
    payment_account_reference: Optional[str] = None
    payment_product_id: Optional[int] = None
    three_d_secure_results: Optional[ThreeDSecureResults] = None
    initial_scheme_transaction_id: Optional[str] = None
    scheme_reference_data: Optional[str] = None
    token: Optional[str] = None
    payment_option: Optional[str] = None
    external_token_linked: Optional[ExternalTokenLinked] = None
    authenticated_amount: Optional[int] = None
    currency_conversion: Optional[CurrencyConversion] = None
    payment_product_3208_specific_output: Optional[PaymentProduct3208SpecificOutput] = None
    payment_product_3209_specific_output: Optional[PaymentProduct3209SpecificOutput] = None
    reattempt_instructions: Optional[ReattemptInstructions] = None